packages/components/src/viewport/withPlaceholder/index.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+import React from 'react';
+import { View } from 'react-native';
+
+export default WrappedComponent => {
+  return class extends React.Component {
+    constructor(props, context) {
+      super(props, context);
+    }
+
+    render() {
+      if (this.props.inViewport) {
+        return <WrappedComponent {...this.props} />;
+      }
+      return <View style={this.props.placeholderStyle} />;
```

Comment:
Inflexible for no good reason. Usage should be either:

```javascript
// placeholder component given during component definition (preferred)
const ImageWithPlaceHolder = WithPlaceHolder(Image, (props, state) => <Text>Placeholder</Text>) 

// or 

const ImageWithPlaceHolder = WithPlaceholder(Image)

// placeholder given as a function parameter at render time
<ImageWithPlaceHolder placeholder={(props, state) => <Text>Placeholder</Text />

Reply:
I think that we will have to go with the second approach. We sometimes want to dynamically create the placeholder, at render time. Here is a real-life example from the news app:
```javascript
render() {
  // calculations of imageSource, newWidth and newHeight

  <ViewportAwareImageWithPlaceholder
    source={imageSource}
    style={{
      width: newWidth,
      height: newHeight,
    }}
    preTriggerRatio={0.5}
    placeHolder={() =>
      <View
        style={{ width: newWidth, height: newHeight, backgroundColor: 'red' }}
      />}
  />
}
```